src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx

```diff
@@ -1,23 +1,39 @@
 import {createStackNavigator} from '@react-navigation/stack';
 import React from 'react';
 import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
 import NoDropZone from '@components/DragAndDrop/NoDropZone';
 import useOnboardingLayout from '@hooks/useOnboardingLayout';
 import useThemeStyles from '@hooks/useThemeStyles';
 import OnboardingModalNavigatorScreenOptions from '@libs/Navigation/AppNavigator/OnboardingModalNavigatorScreenOptions';
+import Navigation from '@libs/Navigation/Navigation';
 import type {OnboardingModalNavigatorParamList} from '@libs/Navigation/types';
 import OnboardingPersonalDetails from '@pages/OnboardingPersonalDetails';
 import OnboardingPurpose from '@pages/OnboardingPurpose';
 import OnboardingWork from '@pages/OnboardingWork';
+import * as Report from '@userActions/Report';
+import ONYXKEYS from '@src/ONYXKEYS';
 import SCREENS from '@src/SCREENS';
 import Overlay from './Overlay';
 
+type OnboardingModalNavigatorProps = {
+    /** Current onboarding completion status */
+    hasCompletedGuidedSetupFlow: boolean;
+};
+
 const Stack = createStackNavigator<OnboardingModalNavigatorParamList>();
 
-function OnboardingModalNavigator() {
+function OnboardingModalNavigator({hasCompletedGuidedSetupFlow}: OnboardingModalNavigatorProps) {
     const styles = useThemeStyles();
     const {shouldUseNarrowLayout} = useOnboardingLayout();
 
+    if (hasCompletedGuidedSetupFlow) {
+        Navigation.goBack();
+        Report.navigateToConciergeChat();
```

Comment:
```suggestion
```

I think just going back without redirecting to Concierge is better. Let me know if there are any edge cases in which redirecting to Concierge is useful.

Reply:
Suppose you are in a report and clicked `https://dev.new.expensify.com:8082/onboarding` link in a message. In this case, if there's only `goBack`,  it will redirect user to onboarding first and then will go back to the current report. So need to redirect user to Concierge again.

What if there's only one `Concierge` redirect? If so, user will be redirect to concierge on deep link click, but if he clicks go back in the browser, it will redirect user to onbarding and then onboarding will redirect user back to concierge, so user can not go back... so these two actions are required and it will remove onboarding from route history...